go/def.bzl

```diff
@@ -625,8 +630,33 @@ go_env_attrs = {
         executable = True,
         single_file = True,
     ),
+    "_test_generator": attr.label(
+        executable = True,
+        default = Label("//go/tools:generate_test_main"),
+        cfg = "host",
+    ),
+    "_extract_package": attr.label(
+        default = Label("//go/tools/extract_package"),
+        executable = True,
+        cfg = "host",
+    ),
 }
 
+
+def get_go_toolchain(ctx):
```

Comment:
I started trying to break up def.bzl and move some of these functions into the private area, which ironically means they need to be public functions, and the name of this is a left over from that, but it did not go well (cgo is a spiderweb), I will try again sometime soon, for now making it private.